src/sys/windows/named_pipe.rs

```diff
@@ -73,6 +73,9 @@ pub struct NamedPipe {
     inner: Arc<Inner>,
 }
 
+unsafe impl Send for NamedPipe {}
+unsafe impl Sync for NamedPipe {}
```

Comment:
Which field in `Inner` is making this necessary?

Reply:
It may make more sense to implement these traits for `Inner` instead, or to wrap the offending field in a newtype and implement them on that newtype.

Reply:
It seems much easier to reason about the specific APIs' usage of `Handle`s, rather than a category of resource backing a `Handle`, for the purpose of justifying `unsafe impl {Send,Sync}`. I'll implement `Send` and `Sync` directly on `iocp::CompletionPort` and `named_pipe::Inner` first.

If there's desire to attempt categorizing types of resources behind `Handle`s as thread-safe, maybe a follow-up PR would be best?